### PR TITLE
FEDX-1399: Added validate publish to checks

### PR DESCRIPTION
**Pull request by matthewnitschke-wk**
# [FEDX-1399](https://jira.atl.workiva.net/browse/FEDX-1399)
Issue Status

Currently its possible to merge a release PR in a state that will fail `dart pub publish`

This PR runs a publish validation for only release PRs so we don't get silent errors

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: .github/workflows/checks.yaml [hunk 1]
```diff
@@ -60,6 +59,17 @@ jobs:
         working-directory: ${{ inputs.package-path }}
         run: dart pub global run dependency_validator
 
+  validate-publish:
+    runs-on: ubuntu-latest
+    # only run on release pull requests
+    if: ${{ startsWith(github.ref, 'refs/head/release') && github.event.sender == 'rmconsole-readonly-wk' }}
```
Comment (evanweible-wf) [inline .github/workflows/checks.yaml:65]: Is it a guarantee that the sender will be `rmconsole-readonly-wk`? I thought we had multiple service accounts and that any of them could be involved based on rate limiting availability

Comment (matthewnitschke-wk) [reply .github/workflows/checks.yaml:65]: Looks like we have 9 of them: https://github.com/orgs/Workiva/people?query=rmconsole

I guess I thought that `readonly` was the one that was used for release prs, although, that might be invalid

We could update that to `contains(github.event.sender, 'rmconsole')`, probably would be a better check

Comment (evanweible-wf) [reply .github/workflows/checks.yaml]: Maybe we can find a different/better way to identify release PRs? Technically, it's possible and totally valid from RM's perspective to manually create a release PR - as long as it meets a couple of criteria, RM will recognize it as such even if it didn't create the PR.

Comment (matthewnitschke-wk) [reply .github/workflows/checks.yaml]: Well, from the perspective of gha-dart-oss, we want to validate for any PR that results in a tag

I'm not sure we'd be able to know that without hitting some, probably non-existant, rmconsole api

An alternate idea would be any PR that changes `pubspec.yaml#version`? would require a bit of introspection into the PR in question though

**Review by evanweible-wf — APPROVED** [review]

**Comment by matthewnitschke-wk** [comment]
QA +1

- [x]  CI does indeed pass

🚀 @Workiva/release-management-p 🚢

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**